design/protected-apps-view.md

```diff
@@ -0,0 +1,917 @@
+const buildPlacementToDRPCMap = (drpcs: DRPlacementControlKind[]) => {
+  const map = {};
+  
+  drpcs.forEach(drpc => {
+    const placementRef = drpc.spec.placementRef;
+    const key = getPlacementUniqueKey(
+      placementRef.name,
+      placementRef.kind,
+      drpc.namespace // Note: DRPC namespace, not Placement namespace (they're the same)
+    );
+    
+    map[key] = drpc;
+  });
+  
+  return map;
+  // Example map:
+  // {
+  //   "payment-service-placement-Placement-openshift-gitops": <DRPC object>,
+  //   "web-frontend-placement-Placement-web-app-ns": <DRPC object>
+  // }
+};
+
+// Step 3: For any Placement, look up its DRPC
+const findDRPCForPlacement = (placement: PlacementKind, drpcMap: PlacementToDRPCMap) => {
+  const key = getPlacementUniqueKey(
+    placement.name,
+    placement.kind,
+    placement.namespace
+  );
+  
+  return drpcMap[key]; // O(1) lookup
+};
+```
+
+This algorithm is elegant and efficient once the map is built. The problem is that in the current frontend implementation, this map must be rebuilt every time any watched resource changes, and the map building requires iterating through all DRPCs in the cluster.
+
+---
+
+## 3. Caveats and Limitations of Current Approach
+
+### 3.1 Performance and Scalability Concerns
+
+#### 3.1.1 Redundant Kubernetes Watches
+Combining all three application types creates 15 separate watch connections to the Kubernetes API:
+ApplicationSet hook: ApplicationSets, Placements, PlacementDecisions, ManagedClusters
+Subscription hook: Applications, Subscriptions, Placements (duplicate), PlacementRules, PlacementDecisions (duplicate)
+DR Resources hook: DRPolicies, DRClusters, DRPlacementControls (all namespaces)
+Discovered hook: DRPlacementControls (openshift-dr-ops namespace - duplicate)
+Each watch maintains a websocket connection streaming all changes cluster-wide, creating significant API server load and network traffic.
+
+#### 3.1.2 Browser-Side Correlation Overhead
+React's useMemo re-runs correlation on every resource change:
+
+New ApplicationSet created → correlate all ApplicationSets
+Placement status changes → correlate all affected applications
+DRPC status updates → correlate all affected applications
+
+The correlation algorithm has O(N*M) complexity (N applications × M DRPCs). With 100 ApplicationSets, 50 Subscriptions, and 200 DRPCs, a single DRPC update triggers tens of thousands of comparisons, causing noticeable UI lag.
+
+#### 3.1.3 Memory Consumption
+Each hook stores its own copy of watched resources. A Placement watched by three hooks exists three times in browser memory. Large clusters consume 50+ MB just for application data, contributing to browser crashes.
+
+#### 3.1.4 Lack of Shared Caching
+useK8sWatchResources maintains per-component state with no cross-component caching. Navigating between pages recreates all watches and refetches all data. Two components needing the same data (e.g., DRPolicies) independently watch and maintain it.
```

Comment:
> useK8sWatchResources maintains per-component state with no cross-component caching.

nits: this is bit misleading, as far as u pass exact same resource object (it "keys" on GVK/name/namespace etc) as an argument, u will only get single duplex (websocket) connection.
If multiple components are using `useK8sWatchResources` and watching exact same resource at exact same time, then all data will be served from single global redux state.

> Navigating between pages recreates all watches and refetches all data.

this is correct.